### PR TITLE
ROX-25948: Decouple ServiceCertificatesRepo from localscanner package

### PR DESCRIPTION
**Pull request by vladbologa**
### Description

https://github.com/stackrox/stackrox/pull/12740 introduces a new Central API to refresh all Secured Cluster certificates. 

On the Sensor side, I plan to reuse parts of the existing certificate refresh logic from the `localscanner` package, which is:
* currently tightly coupled with the local scanner refresh API
* required for backward compatibility with older Central versions, even after the new refresh mechanism is introduced

To prepare for this, I'm extracting the `ServiceCertificatesRepo` from the `localscanner` package, so that it can be used to handle also other sets of certificates, not just local scanner.

Changes in this PR:
* move the `sensor/kubernetes/localscanner` package to `sensor/kubernetes/certrefresh/localscanner`
* extract `ServiceCertificatesRepo` code into ``sensor/kubernetes/certrefresh/certrepo`
* export the interface and implementation so that it can be used by the future cert refresh code
* split the existing `ServiceCertificatesRepo` tests between the two packages (`localscanner` and `certrepo`)

There are no functionality changes in this PR - only refactoring, renames, and the necessary updates to fix compilation and tests. It's probably easiest to review commit by commit.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [x] modified existing tests

#### How I validated my change

Existing tests should verify regressions. There is no new functionality added in this PR.

### CODE REVIEW

File: sensor/kubernetes/certrefresh/localscanner/cert_repo_init.go
Comment (vikin91) [inline sensor/kubernetes/certrefresh/localscanner/cert_repo_init.go:22]: I know the code was doing that the same way before the refactor, but returning an interface instead of a concrete type maybe cumbersome in some situations. A general rule is to accept interfaces on the input but always return the concrete types. Do you think, we could apply this rule here?

**Review by vikin91 — APPROVED** [review]
All looks good! I left one comment to consider.
Please make sure that CI is happy before merging.

File: sensor/kubernetes/certrefresh/localscanner/cert_repo_init.go
Comment (vladbologa) [reply sensor/kubernetes/certrefresh/localscanner/cert_repo_init.go:22]: I suppose the reason it returns the interface is because this function is stored in TLSIssuer as a function pointer:
```go
type serviceCertificatesRepoGetter func(ownerReference metav1.OwnerReference, namespace string,
	secretsClient corev1.SecretInterface) certrepo.ServiceCertificatesRepo
```

I could either:
1. leave as it is
2. change the function pointer to also return a concrete type, but then it couples TLSIssuer with the concrete implementation
3. maybe give up the function pointer approach and just store the returned object - I'm not sure of the implications though yet, I suppose there was a reason for the current design

Comment (vikin91) [reply sensor/kubernetes/certrefresh/localscanner/cert_repo_init.go:22]: Thanks for checking this! 
The decision is up to you. I am fine with any option - including 1.

**Review by porridge — APPROVED** [review]
LGTM modulo the checkboxes in PR description.

**Pull request merged**
